smserver/smserver_test.go

```diff
@@ -207,6 +211,160 @@ func TestBoardConfiguration(t *testing.T) {
 	}
 }
 
+func TestInstanceMessages(t *testing.T) {
+	smConfig := config.Config{
+		SMServerURL: serverURL,
+	}
+
+	testLauncher := &testLauncher{runtimeStatusChan: make(chan launcher.RuntimeStatus, 10)}
+
+	smServer, err := smserver.New(&smConfig, testLauncher, nil, nil, nil, nil, nil, nil,
+		nil, nil, nil, true)
+	if err != nil {
+		t.Fatalf("Can't create: SM Server %s", err)
+	}
+
+	go func() {
+		if err := smServer.Start(); err != nil {
+			t.Errorf("Can't start sm server: %s", err)
+		}
+	}()
+
+	defer smServer.Stop()
+
+	client, err := newTestClient(serverURL)
+	if err != nil {
+		t.Fatalf("Can't create test client: %s", err)
+	}
+
+	defer client.close()
+
+	sendEnvVars := &pb.OverrideEnvVarsRequest{EnvVars: []*pb.OverrideInstanceEnvVar{}}
+	expectedRequest := []cloudprotocol.EnvVarsInstanceInfo{}
+	expectedStatus := &pb.OverrideEnvVarStatus{EnvVarsStatus: []*pb.EnvVarInstanceStatus{}}
+
+	for i := 0; i < 5; i++ {
+		curentTime := time.Now()
+
+		sendEnvVars.EnvVars = append(sendEnvVars.EnvVars, &pb.OverrideInstanceEnvVar{
+			Instance: &pb.InstanceIdent{ServiceId: "id" + strconv.Itoa(i), SubjectId: "s1", Instance: int64(i - 1)},
+			Vars: []*pb.EnvVarInfo{
+				{VarId: "varId" + strconv.Itoa(i), Variable: "variable " + strconv.Itoa(i)},
+				{VarId: "varId2" + strconv.Itoa(i), Variable: "variable 2" + strconv.Itoa(i), Ttl: timestamppb.New(curentTime)},
+			},
+		})
+
+		sendEnvVars.EnvVars[0].Instance.SubjectId = ""
+
+		expectedRequest = append(expectedRequest, cloudprotocol.EnvVarsInstanceInfo{
+			InstanceFilter: createInstanceFilter("id"+strconv.Itoa(i), "s1", int64(i-1)),
+			EnvVars: []cloudprotocol.EnvVarInfo{
+				{ID: "varId" + strconv.Itoa(i), Variable: "variable " + strconv.Itoa(i)},
+				{ID: "varId2" + strconv.Itoa(i), Variable: "variable 2" + strconv.Itoa(i), TTL: &curentTime},
+			},
+		})
+
+		expectedStatus.EnvVarsStatus = append(expectedStatus.EnvVarsStatus, &pb.EnvVarInstanceStatus{
+			Instance: &pb.InstanceIdent{ServiceId: "id" + strconv.Itoa(i), SubjectId: "s1", Instance: int64(i - 1)},
+			VarsStatus: []*pb.EnvVarStatus{
+				{VarId: "varId" + strconv.Itoa(i)}, {VarId: "varId2" + strconv.Itoa(i)},
+			},
+		})
+	}
+
+	sendEnvVars.EnvVars[0].Instance.SubjectId = ""
+	expectedRequest[0].InstanceFilter.SubjectID = nil
+	expectedStatus.EnvVarsStatus[0].Instance.SubjectId = ""
+
+	envVarStatus, err := smServer.OverrideEnvVars(context.Background(), sendEnvVars)
+	if err != nil {
+		t.Fatalf("Can't override env vars: %s", err)
+	}
+
+	if !proto.Equal(envVarStatus, expectedStatus) {
+		t.Errorf("Incorrect overrode env var status")
+	}
+
+	if reflect.DeepEqual(expectedRequest, testLauncher.envVarRequest) {
```

Comment:
a little confusing this check, we created the variable `expectedRequest` but in fact we have the opposite. But up to you

Reply:
tests also performs validation of expected request